src/cmd/tools.go

```diff
@@ -0,0 +1,131 @@
+package cmd
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/opslevel/cli/common"
+	"github.com/opslevel/opslevel-go/v2024"
+	"github.com/spf13/cobra"
+)
+
+var createServiceToolCmd = &cobra.Command{
+	Use:   "tool",
+	Short: "Create service tool",
+	Example: `
+cat << EOF | opslevel create service tool -f -
+service: my-service-alias
+category: deployment
+displayName: "fancy tool"
+environment: "dev"
+url: "https://example.com"
+EOF
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		client := getClientGQL()
+		serviceIdentifier := readServiceFieldFromYaml()
+		toolFromInput, err := readResourceInput[opslevel.ToolCreateInput]()
+		cobra.CheckErr(err)
+
+		err = verifyToolCategory(&toolFromInput.Category)
+		cobra.CheckErr(err)
+
+		service, err := getService(serviceIdentifier)
+		cobra.CheckErr(err)
+
+		toolFromInput.ServiceId = &service.Id
+		tool, err := client.CreateTool(*toolFromInput)
+		cobra.CheckErr(err)
+		common.PrettyPrint(string(tool.Id))
+	},
+}
+
+var updateServiceToolCmd = &cobra.Command{
+	Use:     "tool",
+	Aliases: []string{"tool"},
+	Short:   "Update service tool",
+	Args:    cobra.ExactArgs(1),
+	Example: `
+cat << EOF | opslevel update service tool tool-ID -f -
+service: my-service-alias
+category: deployment
+displayName: "fancy tool"
+environment: "dev"
+url: "https://example.com"
+EOF
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		id := args[0]
+		if !opslevel.IsID(id) {
+			cobra.CheckErr(fmt.Errorf("invalid ID: '%s'", id))
+		}
+		toolId := opslevel.ID(id)
+
+		serviceIdentifier := readServiceFieldFromYaml()
+		toolFromInput, err := readResourceInput[opslevel.ToolUpdateInput]()
+		cobra.CheckErr(err)
+
+		err = verifyToolCategory(toolFromInput.Category)
+		cobra.CheckErr(err)
+
+		client := getClientGQL()
+		service, err := getService(serviceIdentifier)
+		cobra.CheckErr(err)
+
+		if !isToolIdInServiceTools(toolId, service.Tools) {
```

Comment:
This feels like unnecessary validation - the API will do this for us AND this incurs extra API calls - service get which we just found out is heavy

Reply:
😬 yes this needs to go. I think this was a product of my fevered thinking, literally